src/utils/defaults.ts

```diff
@@ -29,10 +29,9 @@ function findFile(file: string, preserveSymlinks: boolean): string | void {
 function addJsExtensionIfNecessary(file: string, preserveSymlinks: boolean) {
 	let found = findFile(file, preserveSymlinks);
 	if (found) return found;
-	for (let ext of jsExts) {
-		found = findFile(file + ext, preserveSymlinks);
-		if (found) return found;
-	}
+	found = findFile(file + '.js', preserveSymlinks);
+	if (found) return found;
+	return findFile(file + '.mjs', preserveSymlinks);
```

Comment:
But going back to my original comment, wouldn't this allow `.mjs` to be used without an extension while giving precedence to `.js` in these situations which is probably *not* in rollup's best interest? If we do this, I think these two extensions + their precedence should be configurable as an array with the default being `['mjs', 'js']`. On the other hand, this would FINALLY settle some long-standing feature requests, cf #1894

Reply:
I forgot we were supporting mjs here actually, also only just saw this again when refactoring the above. I've fixed this up to at least check in the right order, meaning yes we can properly support .mjs now!

Personally, I don't think we should have an array here - anything that configures the resolver should also configure the loader to support the translate for that file extension. `.mjs` and `.js` are the only valid registered MIMES for JavaScript files, with a well-defined precedence, and anything else will be a different module format.

Reply:
Makes sense 👍 I still think that we may want to configure this at some point (if people ask about it?) but with this behaviour in place, this can definitely wait for another PR. I just remember struggling with some linter settings around this behaviour at work and being happy that Webpack has an option to configure just that.